src/wallet/signer.rs

```diff
@@ -569,6 +569,21 @@ mod signers_container_tests {
     use miniscript::ScriptContext;
     use std::str::FromStr;
 
+    // Return true if data field of `fat` pointer is the same address as `thin` pointer.
+    fn is_equal(fat: &Arc<dyn Signer>, thin: &Arc<DummySigner>) -> bool {
+        let (left, right) = raw_pointers(fat, thin);
+        left == right
+    }
```

Comment:
I don't think this part is necessary, you should be able to just compare two signers using their `.id()`.

I would change `DummySigner` to take a `u64` inside, and then return it as its id with something like `SignerId::Dummy(self.number)`.

Then if you create different signers with different values you can compare them using the id.

Reply:
Cool, nice idea. One thing was a bit ugly, I had to use a helper function instead of implementing `PartialEq` since we only have a reference to a `dyn Signer` when comparing for equality. Please let me know if you see a cleaner way to do this.